### PR TITLE
[SPARK-7013][ML][test]Add unit test for spark.ml StandardScaler

### PR DESCRIPTION
**Pull request by RoyGao**
I have added unit test for ML's StandardScaler By comparing with R's output, please review  for me.
Thx.

### CODE REVIEW

File: mllib/src/test/scala/org/apache/spark/ml/feature/StandardScalerSuite.scala [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+package org.apache.spark.ml.feature
+
+import org.scalatest.FunSuite
+
+import org.apache.spark.mllib.linalg.{DenseVector, SparseVector, Vector, Vectors}
+import org.apache.spark.mllib.util.MLlibTestSparkContext
+import org.apache.spark.mllib.util.TestingUtils._
+import org.apache.spark.sql.{DataFrame, Row, SQLContext}
+
+class StandardScalerSuite extends FunSuite with MLlibTestSparkContext{
```
Comment (mengxr) [inline mllib/src/test/scala/org/apache/spark/ml/feature/StandardScalerSuite.scala]: Please extend `SparkFunSuite` instead (we added some tools to help Jenkins logging).

File: mllib/src/test/scala/org/apache/spark/ml/feature/StandardScalerSuite.scala [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+      .fit(dataFrame)
+
+    val standardscaler3 = new StandardScaler()
+      .setInputCol("features")
+      .setOutputCol("standarded_features")
+      .setWithMean(false)
+      .setWithStd(false)
+      .fit(dataFrame)
+
+    val res1 = collectResult(standardscaler1.transform(dataFrame))
```
Comment (mengxr) [inline mllib/src/test/scala/org/apache/spark/ml/feature/StandardScalerSuite.scala]: `collect` doesn't guarantee the ordering (if there exists multiple partitions). So a better approach would be put the expected result side by side in the same row, and then compare them. See

https://github.com/apache/spark/blob/master/mllib/src/test/scala/org/apache/spark/ml/feature/Word2VecSuite.scala#L48

File: mllib/src/test/scala/org/apache/spark/ml/feature/StandardScalerSuite.scala [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+
+    data = Array(
+      Vectors.dense(-2.0, 2.3, 0.0),
+      Vectors.dense(0.0, -1.0, -3.0),
+      Vectors.dense(0.0, -5.1, 0.0),
+      Vectors.dense(3.8, 0.0, 1.9),
+      Vectors.dense(1.7, -0.6, 0.0),
+      Vectors.dense(0.0, 1.9, 0.0)
```
Comment (mengxr) [inline mllib/src/test/scala/org/apache/spark/ml/feature/StandardScalerSuite.scala]: Please keep the unit test minimal. If there is no particular reason to include 6 examples, we can have less.

**Comment by mengxr** [comment]
ok to test

File: mllib/src/test/scala/org/apache/spark/ml/feature/StandardScalerSuite.scala [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+      Vectors.dense(0.0, -1.367762550529, 0.590968109266),
+      Vectors.dense(0.917913720165, -0.160913241239, 1.950194760579)
+    )
+    resWithBoth = Array(
+      Vectors.dense(-1.0259035695965, 0.920781324866, -0.8470542899497),
+      Vectors.dense(0.0539949247156, -1.063815317078, -0.256086180682),
+      Vectors.dense(0.9719086448809, 0.143033992212, 1.103140470631)
+    )
+  }
+
```
Comment (hhbyyh) [inline mllib/src/test/scala/org/apache/spark/ml/feature/StandardScalerSuite.scala]: checkParams?

**Comment by mengxr** [comment]
ok to test

**Comment by mengxr** [comment]
LGTM pending Jenkins.

File: mllib/src/test/scala/org/apache/spark/ml/feature/StandardScalerSuite.scala (hunk 4 above)
Comment (mengxr) [inline mllib/src/test/scala/org/apache/spark/ml/feature/StandardScalerSuite.scala]: We can add it in a separate PR.

**Comment by mengxr** [comment]
Merged into master and branch-1.6. Thanks!

**Pull request closed without merging**